packages/nx-plugin/src/generators/application/application.unit.test.ts

```diff
@@ -0,0 +1,86 @@
+import { readProjectConfiguration, Tree } from '@nx/devkit';
+import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
+import { AngularApplicationSchema } from './schema';
+import applicationGenerator from './application';
+import { describe, it, expect, beforeEach } from 'vitest';
+
+describe('Application Generator', () => {
+  let tree: Tree;
+
+  beforeEach(() => {
+    tree = createTreeWithEmptyWorkspace();
+  });
+
+  it('should throw if the directory and name is not present in the options', async () => {
+    await expect(() => applicationGenerator(tree, {} as any)).rejects.toThrow(
+      'You must provide a directory where the application will be placed.'
+    );
+  });
+
+  it('should throw if the directory is the current working directory and name is not present in the options', async () => {
+    await expect(() =>
+      applicationGenerator(tree, { directory: '.' } as any)
+    ).rejects.toThrow(
+      'When placing an application in the current working directory, you must provide a name for the application with --name'
+    );
+  });
+
+  it.todo('should generate an application correctly', async () => {
+    const options: AngularApplicationSchema = {
+      directory: 'apps/demo',
+      e2eTestRunner: 'none',
+      skipPackageJson: true,
+    };
+
+    // @TODO applicationGenerator should not throw
+    await expect(applicationGenerator(tree, options)).resolves.not.toThrow();
```

Comment:
This line was implemented as `await applicationGenerator(tree, options)` and by wrapping it with a check to not throw it got visible that it did not work in the past properly.